packages/testcontainers/src/generic-container/generic-container-reuse.test.ts

```diff
@@ -102,6 +110,40 @@ describe("GenericContainer reuse", () => {
     await container2.stop();
   });
 
+  it("should reuse container when an existing reusable container has stopped but not removed", async () => {
+    const container1 = await new GenericContainer("cristianrgreco/testcontainer:1.1.14")
+      .withName("there_can_only_be_one")
+      .withExposedPorts(8080)
+      .withReuse()
+      .start();
+    await container1.stop({ remove: false });
+
+    // Wait for the container to be stopped, seems to be asynchronous
+    const status = await new IntervalRetry<string, string>(10).retryUntil(
+      () =>
+        client.container
+          .getById(container1.getId())
+          .inspect()
+          .then((i) => i.State.Status)
+          .catch(() => "failed"),
+      (status) => status === "exited",
+      () => "timeout",
+      500
+    );
+
+    expect(status).toBe("exited");
```

Comment:
Just to explain this one. I'm unsure if I'm hitting on a race condition here, but without this check it seems like the `container1`'s `Status` is still `running` while trying to start `container2`. In other words, just awaiting `container1.stop({ remove: false })` seems to not be enough. 

If I set the status filter in `fetchByLabel` to just filter `running` containers, then the test is still green without this `IntervalRetry` status-check. But the test becomes red if I do this check (which it should do, because I'm testing for stopped containers).

Reply:
> In other words, just awaiting container1.stop({ remove: false }) seems to not be enough.

You'll need to give a timeout to let Testcontainers wait for the container to be stop. By default the timeout is `0`, meaning Testcontainers will issue the stop command and move on (which is why immediately after the container is still in a `running` state

Reply:
Oh, I've read the code wrong then, sorry 🙈 It seemed to just pipe the timeout value to `dockerode` which I thought was just to give the container a chance to gracefully shut down 😅

Reply:
That's waiting for 10ms

Reply:
Oh sorry, I just followed the code to here: https://github.com/testcontainers/testcontainers-node/blob/main/packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts#L128

Which lead to `dockerode`, and there `t` said "number of seconds" 😅 :

```
    interface ContainerStopOptions {
        signal?: string;
        /** Number of seconds to wait before killing the container */
        t?: number;
        abortSignal?: AbortSignal;
    }
```

Upping it to `10000` fixes it, thanks 👍🏻